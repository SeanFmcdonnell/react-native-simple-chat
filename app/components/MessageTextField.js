import React from 'react';
import { StyleSheet, TextInput } from 'react-native';

class MessageTextField extends React.Component {

    constructor(props) {
        super(props);
        this.onSubmit = this.props.onSubmit;
        this.state = { value: null };
    }

    render() {
        return (
            <TextInput
                placeholder='Message'
                value={ this.state.value }
                style={ styles.textInput }
                onChangeText={ (text) => this.setState({ value: text }) }
                onSubmitEditing={ (event) => this.onSubmit(event.nativeEvent.text, () => this.setState({ value: null })) }/>
        );
    }
}

const styles = StyleSheet.create({
    textInput: {
        textAlign: 'left',
        paddingLeft: 10,
        height: 35,
        borderWidth: 1,
        borderColor: '#EEEEEE',
        borderRadius: 20,
        backgroundColor: "#ffffff"
    }
});

export {
    MessageTextField
};
