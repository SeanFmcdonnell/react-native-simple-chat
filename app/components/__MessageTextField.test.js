import React from 'react';

import renderer from 'react-test-renderer';
import { MessageTextField } from "./MessageTextField";

test('renders correctly', () => {
    var text = "hello";
    const tree = renderer.create(<MessageTextField onSubmit={ (message, cb) => {
    } }/>).toJSON();
    expect(tree).toMatchSnapshot();
});
