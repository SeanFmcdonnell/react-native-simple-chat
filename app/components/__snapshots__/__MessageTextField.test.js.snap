// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<TextInput
  allowFontScaling={true}
  onChangeText={[Function]}
  onSubmitEditing={[Function]}
  placeholder="Message"
  style={
    Object {
      "backgroundColor": "#ffffff",
      "borderColor": "#f5f5f5",
      "borderRadius": 20,
      "borderWidth": 1,
      "height": 35,
      "paddingLeft": 10,
      "textAlign": "left",
    }
  }
  value={null}
/>
`;
